ulam-spiral/src/main.rs

```diff
@@ -14,8 +14,90 @@
 // limitations under the License.
 //
 
+use std::ops::{Index, IndexMut};
 use image::{ImageBuffer, Luma};
 
+struct SquareMatrix {
+    elems: Vec<usize>
+}
+
+impl SquareMatrix {
+    pub fn new(size: usize) -> Self {
+        assert!(size > 0 && size % 2 != 0, "Matrix size must be odd and bigger than 0!");
+
+        let mut elems = vec![0; size * size];
+
+        let mut dir: Directions = Directions::RIGHT;
+        let mut step: usize = 1;
+        let mut xy_cursor: (usize, usize) = (size / 2, size / 2);
+        elems[(xy_cursor.1 * size) + xy_cursor.0] = 1;
+
+        let mut i = 2;
+        'outer: loop {
```

Comment:
Hey! Cheater! Your type is named "Square Matrix", not "UlamSpiral", why does its constructor do the spiral part?